system/Email/Email.php

```diff
@@ -2019,45 +2012,73 @@ protected function SMTPAuthenticate()
             return true;
         }
 
-        if ($this->SMTPUser === '' && $this->SMTPPass === '') {
+        // If no username or password is set
+        if ($this->SMTPUser === '' || $this->SMTPPass === '') {
             $this->setErrorMessage(lang('Email.noSMTPAuth'));
 
             return false;
         }
 
-        $this->sendData('AUTH LOGIN');
-        $reply = $this->getSMTPData();
+        switch ($this->SMTPAuthMethod) {
+            case 'LOGIN':
+                $this->sendData('AUTH LOGIN');
+                $reply = $this->getSMTPData();
 
-        if (str_starts_with($reply, '503')) {    // Already authenticated
-            return true;
-        }
+                if (str_starts_with($reply, '503')) {    // Already authenticated
+                    return true;
+                }
 
-        if (! str_starts_with($reply, '334')) {
-            $this->setErrorMessage(lang('Email.failedSMTPLogin', [$reply]));
+                if (! str_starts_with($reply, '334')) {
+                    $this->setErrorMessage(lang('Email.failedSMTPLogin', [$reply]));
 
-            return false;
-        }
+                    return false;
+                }
 
-        $this->sendData(base64_encode($this->SMTPUser));
-        $reply = $this->getSMTPData();
+                $this->sendData(base64_encode($this->SMTPUser));
+                $reply = $this->getSMTPData();
 
-        if (! str_starts_with($reply, '334')) {
-            $this->setErrorMessage(lang('Email.SMTPAuthUsername', [$reply]));
+                if (! str_starts_with($reply, '334')) {
+                    $this->setErrorMessage(lang('Email.SMTPAuthUsername', [$reply]));
 
-            return false;
-        }
+                    return false;
+                }
 
-        $this->sendData(base64_encode($this->SMTPPass));
-        $reply = $this->getSMTPData();
+                $this->sendData(base64_encode($this->SMTPPass));
+                $reply = $this->getSMTPData();
 
-        if (! str_starts_with($reply, '235')) {
-            $this->setErrorMessage(lang('Email.SMTPAuthPassword', [$reply]));
+                if (! str_starts_with($reply, '235')) {
+                    $this->setErrorMessage(lang('Email.SMTPAuthPassword', [$reply]));
 
-            return false;
+                    return false;
+                }
+                break;
+
+            case 'PLAIN':
+                // Generate single command for PLAIN authentication
+                $authString = "\0" . $this->SMTPUser . "\0" . $this->SMTPPass;
+
+                $this->sendData('AUTH PLAIN ' . base64_encode($authString));
```

Comment:
Apparently, the single-line auth is not widely supported. I tested it on two different providers and it failed. A safer option will be to require a challenge.
```
C: AUTH PLAIN
S: 334
C: base64 string
S: 235
```